server/src/main/java/org/cloudfoundry/identity/uaa/oauth/token/JwtTokenGranter.java

```diff
@@ -13,11 +13,18 @@
 import org.cloudfoundry.identity.uaa.oauth.common.exceptions.InvalidGrantException;
 import org.cloudfoundry.identity.uaa.oauth.provider.ClientDetails;
 
+import java.util.List;
+
+import static org.cloudfoundry.identity.uaa.oauth.token.TokenConstants.CLIENT_AUTH_PRIVATE_KEY_JWT;
+import static org.cloudfoundry.identity.uaa.oauth.token.TokenConstants.CLIENT_AUTH_SECRET;
 import static org.cloudfoundry.identity.uaa.oauth.token.TokenConstants.GRANT_TYPE_JWT_BEARER;
 
 public class JwtTokenGranter extends AbstractTokenGranter {
     final DefaultSecurityContextAccessor defaultSecurityContextAccessor;
 
+    private static final List<String> ALLOWED_AUTH_METHODS = List.of(CLIENT_AUTH_SECRET, CLIENT_AUTH_PRIVATE_KEY_JWT);
+            //comment I would add here CLIENT_AUTH_EMPTY to allow empty secret for jwt bearer, same as allowed for password
```

Comment:
With this, the change is not change but only a refactoring.

As mentioned currently we reject in clients for JWT bearer empty secret but that is strange and we should discuss